tests/functional_tests/python_test_utils/test_grpo_training_loop.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+
+import json
+import logging
+import math
+from statistics import median
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def test_grpo_training_loop(golden_values_path: str, test_values_path: str) -> None:
+
+    with open(golden_values_path, 'r') as f1, open(test_values_path, 'r') as f2:
+        golden_values_content = f1.read()
+        tensorboard_content = f2.read()
+
+    output_groundtruth = json.loads(golden_values_content)
+
+    if isinstance(output_groundtruth, str):
+        # Handle JSONL output, assume only one line in this case.
+        output_groundtruth = json.loads(output_groundtruth)
+
+    output_current = json.loads(tensorboard_content)
+    if isinstance(output_current, str):
+        # Handle JSONL output, assume only one line in this case.
+        output_current = json.loads(output_current)
+
+    assert set(output_groundtruth.keys()).issuperset(
+        set(output_current.keys())
+    ), f"Some IDs from groundtruth are missing in current: {output_groundtruth.keys()} vs {output_current.keys()}"
+    if set(output_groundtruth.keys()) != set(output_current.keys()):
+        logger.warning(
+            f"Some IDs from groundtruth are missing in output, only the subset of ids in groundtruth will be tested: {output_groundtruth.keys()} vs {output_current.keys()}"
+        )
+    assert len(output_groundtruth) > 0, "No test performed for output"
+
+    # iteration-time assertions.
+    if "iteration-time" in output_groundtruth.keys():
+
+        # First warmup iteration is excluded from iteration-time statistics.
+        iteration_time_sampled = median([l for l in output_current["iteration-time"]['values'].values()][1:])
+        iteration_time_golden = median([l for l in output_groundtruth["iteration-time"]['values'].values()][1:])
+
+        # 10% is empirically observed to be within hardware variance.
+        assert (
+            iteration_time_sampled >= 0.9 * iteration_time_golden
```

Comment:
I think this is wrong. Imagine golden time is 100 secs, and sampled time 50 seconds (2x faster). 50>=90 will trigger the assert, which shouldn't be the case. We should trigger the next one with this time.

Reply:
fixed - please review again